.github/workflows/main.yml

```diff
@@ -0,0 +1,253 @@
+        with:
+          name: Test Results
+          path: "**/*-test-results.trx"
+          reporter: dotnet-trx
+          fail-on-error: true
+
+  # Publish the apps
+  pack:
+    runs-on: ubuntu-latest
+    permissions:
+      actions: write
+      contents: read
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+
+      - name: Install .NET
+        uses: actions/setup-dotnet@607fce577a46308457984d59e4954e075820f10a # v3.0.3
+
+      - name: Run restore
+        run: dotnet restore
+
+      - name: Run build
+        run: >
+          dotnet build
+          --no-restore
+          --configuration Release
+          -p:ContinuousIntegrationBuild=true
+
+      # When triggered by `push` or `pull_request` events, generate a prerelease version
+      # for the apps, so as to clearly indicate that it's not a stable version release.
+      - name: Generate prerelease version
+        id: prerelease-version
+        if: ${{ github.event_name != 'release' }}
+        run: |
+          time=$(date +%s)
+          ref="${{ github.head_ref || github.ref_name }}"
+          ref_clean="${ref/\//-}"
+          suffix="ci-$time-$ref_clean-${{ github.run_id }}"
+          echo "version=0.0.0-$suffix" >> $GITHUB_OUTPUT
+
+      - name: Publish API
+        run: >
+          dotnet publish src/Api/
+          --no-restore
+          --no-build
+          --configuration Release
+          --runtime win-x86
+          --no-self-contained
+          --output publish/Api/
+          -p:ContinuousIntegrationBuild=true
+          ${{ steps.prerelease-version.outputs.version && format('-p:Version={0}', steps.prerelease-version.outputs.version) || '' }}
+
+      - name: Publish AdminConsole
+        run: >
+          dotnet publish src/AdminConsole/
+          --no-restore
+          --no-build
+          --configuration Release
+          --runtime win-x86
+          --no-self-contained
+          --output publish/AdminConsole/
+          -p:ContinuousIntegrationBuild=true
+          ${{ steps.prerelease-version.outputs.version && format('-p:Version={0}', steps.prerelease-version.outputs.version) || '' }}
+
+      - name: Upload API artifacts
+        uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32 # v3.1.3
+        with:
+          name: Api
+          path: publish/Api/
+
+      - name: Upload AdminConsole artifacts
+        uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32 # v3.1.3
+        with:
+          name: AdminConsole
+          path: publish/AdminConsole/
+
+  # Deploy the apps
+  deploy:
+    needs:
+      # Technically, it's not required for the format job to succeed for us to push the package,
+      # so we may consider removing it as a prerequisite here.
+      - format
+      - test
+      - pack
+    runs-on: ubuntu-latest
+    environment: devtest
```

Comment:
One environment for both projects is used here. Currently, we have one for each project. Not sure if there's a lot of reason for that, given that (from what I understand) both API and AdminConsole are deployed in tandem. Open for input.

Reply:
It's fine. However, I think the secrets may differ between API/AdminConsole, so we would need to prefix their name with api/adminconsole